src/NetTopologySuite/Operation/Linemerge/LineSequencer.cs

```diff
@@ -480,8 +480,9 @@ private Geometry BuildSequencedGeometry(IEnumerable<IEnumerable<DirectedEdge>> s
 
         private static LineString Reverse(LineString line)
         {
-            var pts = line.Coordinates;
-            Array.Reverse(pts);
+            // Use a new Coordinate Array to avoiding reversing the Coordinates of the line
+            var pts = line.CoordinateSequence.Reversed().ToCoordinateArray();
```

Comment:
`ToCoordinateArray()` might make an extra copy, and then `CreateLineString(Coordinate[])` might make yet another extra one, both depending on what the geometry factory uses for its `CoordinateSequenceFactory`.

Suggestion is to use `CreateLineString(CoordinateSequence)` directly from the result of `line.CoordinateSequence.Reversed()`

Reply:
Done